### PR TITLE
Allow number of buffers to be set, and allow frame buffer to be explicitly released.

### PR DESCRIPTION
**Pull request by aamcrae**
Some changes to allow frame grabbing to be more up to date.
 - Allow a settable number of frame buffers.
 - Allow a frame to be separately released back to the buffer queue so that it does not get reused.

### CODE REVIEW

File: webcam.go [hunk 1]
```diff
@@ -164,21 +175,34 @@ func (w *Webcam) StartStreaming() error {
...
+        w.ReleaseFrame(index)
+    }
+    return result, err
+}
+
+// Get a single frame from the webcam and return the frame and
+// the buffer index. To return the buffer, ReleaseFrame must be called.
+// If frame cannot be read at the moment
+// function will return empty slice
+func (w *Webcam) GetFrame() ([]byte, uint32, error) {
```
Comment (blackjack) [inline webcam.go]: Is there a use case for this?

File: webcam.go [hunk 2]
```diff
@@ -113,22 +117,29 @@ func (w *Webcam) SetImageFormat(f PixelFormat, width, height uint32) (PixelForma
 	}
 }
 
+// Set the number of frames to be buffered.
+func (w *Webcam) SetBufferCount(count uint32) error {
+    if count < 2 || count > maxBufferCount {
```
Comment (blackjack) [inline webcam.go]: Probably you should not limit a lower bound as well, V4l2 documentation does not mention neither lower nor upper limit on buffer count so I suggest to leave it to developer. 

Also this function does not have any effect if streaming already started, so I suggest to mention it in a doc.

Finally, according to v4l2 docs, user can actually request more buffers even if streaming already started, so as extra step would be nice to provide this feature (optionally, if you have time)

File: webcam.go (hunk 1 above)
Comment (aamcrae) [reply webcam.go]: The primary case (and the main reason I suggested these changes) is to allow the grabbing of the most recent frames possible (i.e a small buffer count) and to allow some potentially lengthy image processing on them - I found that if the buffer was released back to the pool immediately, the camera could overwrite the buffer contents.

File: webcam.go [hunk 3]
```diff
@@ -13,9 +13,12 @@ import (
 // Webcam object
 type Webcam struct {
 	fd      uintptr
+    bufcount uint32
```
[the remark this answers is not in the document]
Comment (aamcrae) [reply webcam.go]: Done

File: webcam.go [hunk 4]
```diff
@@ -13,9 +13,12 @@ import (
...
 	buffers [][]byte
 }
 
+const maxBufferCount = 512
```
[the remark this answers is not in the document]
Comment (aamcrae) [reply webcam.go]: Done

File: webcam.go (hunk 2 above)
Comment (aamcrae) [reply webcam.go]: Doing it on the fly seems tricky, so perhaps allowing it via stopping streaming, then changing the buffer count, then starting again seems a cleaner approach.

File: webcam.go (hunk 1 above)
Comment (aamcrae) [reply webcam.go]: ping?

Comment (blackjack) [reply webcam.go]: Sorry for late response. I'll finish the review today

**Comment by blackjack** [comment]
Thank you for your effort, sorry for being not very active.

**Pull request merged**